nodeAPI/src/classes/deleteClass.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+import { convertStringToArray } from '../models/utilities';
+import { convertArrayToString } from '../models/utilities';
+
+/**
+ * Gets classes based on classid.
+ * @param  {Request} req  body: { classid: string }
+ * @param  {Response} res
+ * @param  {postgres.Pool} pool
+ * @return {Promise}  status: 200, 404, 500
+ */
+export const request = async (req, res, pool) => {
+  try {
+    let classID = req.body.classid;
+    let query = {
+      text: 'SELECT * FROM classes WHERE classid = $1',
+      values: [classID]
+    };
+    let classToDelete = await pool.query(query);
+
+    let response;
+    if (classToDelete.rows.length > 0) {
+      res.status(200);
+
+      let users = convertStringToArray(classToDelete.row[0].studentids);
+      users.push(classToDelete.row[0].instructorID);
+      for(let i=0; i<users.length; i++)
+      {
+        let query2 = {
+          text: 'SELECT userclasses FROM users WHERE userid = $1',
+          values: [users[i]]
+        };
+        let userClasses = convertStringToArray(await pool.query(query2).row[0]);
+        userClasses.splice( userClasses.indexOf(classID), 1 );
```

Comment:
What is this for?

Reply:
delete the class from the user's class list

Reply:
userClasses is now an array so splice isn't needed.

Reply:
no u